_posts/news/2022-04-04-announcing-scalajs-1.10.0.md

```diff
@@ -0,0 +1,95 @@
+To address that, Scala.js 1.10.x uses `java.security.SecureRandom` in its implementation instead.
+
+However, `SecureRandom` cannot be implemented on all JavaScript environments, as ECMAScript, per se, lacks any primitive to access a good source of entropy.
+While it can be implemented on top of browser APIs or Node.js APIs, we have no decent fallback for other cases.
+Not fully implementing it correctly for all platforms would go against our policy for the standard library, i.e., that linking code must be correct.
+Therefore, Scala.js core does not provide any implementation of `java.security.SecureRandom`.
+
+This means that code that was previously calling `java.util.UUID.randomUUID()` will now fail to link.
+To preserve binary compatibility, we introduce two variants of a library that provides `java.security.SecureRandom`:
+
+* [`scalajs-java-securerandom`](https://github.com/scala-js/scala-js-java-securerandom) provides a *correct*, cryptographically secure implementation of `java.security.SecureRandom`, but relies on the Node.js `crypto` package or the Web Crypto API `crypto.getRandomValues` to be available (e.g., in browsers).
+* [`scalajs-fake-insecure-securerandom`](https://github.com/scala-js/scala-js-fake-insecure-securerandom) is a fake, *insecure* implementation that works in any ECMAScript environment. The only reason this exists is to unblock migration from Scala.js 1.9.x and earlier, in situations that require `randomUUID()` and can accept that they will be insecure.
+
+Due to the changes in the core library, you may encounter linking errors when upgrading to Scala.js 1.10.0, such as:
+
+{% highlight text %}
+[error] Referring to non-existent class java.security.SecureRandom
+[error]   called from private java.util.UUID$.csprng$lzycompute()java.util.Random
+[error]   called from private java.util.UUID$.csprng()java.util.Random
+[error]   called from java.util.UUID$.randomUUID()java.util.UUID
+[error]   called from static java.util.UUID.randomUUID()java.util.UUID
+[error]   called from helloworld.HelloWorld$.main([java.lang.String)void
+[error]   called from static helloworld.HelloWorld.main([java.lang.String)void
+[error]   called from core module module initializers
+[error] involving instantiated classes:
+[error]   java.util.UUID$
+[error]   helloworld.HelloWorld$
+{% endhighlight %}
+
+You may fix these linking errors by adding the following dependency to your `libraryDependencies`:
+
+{% highlight scala %}
+"org.scala-js" %%% "scalajs-java-securerandom" % "1.0.0"
+{% endhighlight %}
+
+As mentioned above, this will only work in environments that either provide the Web Crypto API, or Node.js' `crypto` package.
+
+If you need random UUID generation in other environments, we encourage you to implement it yourselves, in a way that corresponds to your requirements in terms of security and collision likelihood.
```

Comment:
I think it is good that we try to provide guidance that goes beyond "use fake insecure otherwise". However, wouldn't the next step here be something like: "Find if your environment has a cryptographic random primitive and PR it to `scalajs-java-securerandom`"? Maybe with an intermediate "shim it onto `crypto.getRandomValues`".